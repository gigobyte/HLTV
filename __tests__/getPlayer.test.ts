import HLTV from '../src/'

function sleep(ms: number): Promise<void> {
  return new Promise(resolve => setTimeout(resolve, ms))
}

test('getPlayer', async () => {
  await sleep(1000)
  expect(await HLTV.getPlayer({ id: 8000 })).toMatchSnapshot()
  expect(await HLTV.getPlayer({ id: 7000 })).toMatchSnapshot()
  await sleep(3000)
  expect(await HLTV.getPlayer({ id: 3000 })).toMatchSnapshot()
  expect(await HLTV.getPlayer({ id: 9123 })).toMatchSnapshot()
  await sleep(1000)
}, 30000)
