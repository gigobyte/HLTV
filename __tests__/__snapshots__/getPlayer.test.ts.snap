// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPlayer 1`] = `
Object {
  "achievements": Array [],
  "age": 23,
  "country": Object {
    "code": "DE",
    "name": "Germany",
  },
  "facebook": undefined,
  "id": 8000,
  "ign": "syken",
  "image": "https://static.hltv.org/images/playerprofile/thumb/8000/400.jpeg?v=5",
  "name": "Thilo Phan",
  "statistics": Object {
    "deathsPerRound": 0.72,
    "headshots": 58.8,
    "killsPerRound": 0.65,
    "mapsPlayed": 4,
    "rating": 0.9,
    "roundsContributed": 62.9,
  },
  "team": undefined,
  "twitch": undefined,
  "twitter": undefined,
}
`;

exports[`getPlayer 2`] = `
Object {
  "achievements": Array [],
  "age": undefined,
  "country": Object {
    "code": "DE",
    "name": "Germany",
  },
  "facebook": undefined,
  "id": 7000,
  "ign": "Allanon",
  "image": "https://static.hltv.org/images/playerprofile/blankplayer.svg",
  "name": "Sebastian Orth",
  "statistics": Object {
    "deathsPerRound": 0,
    "headshots": 0,
    "killsPerRound": 0,
    "mapsPlayed": 0,
    "rating": 0,
    "roundsContributed": 0,
  },
  "team": undefined,
  "twitch": undefined,
  "twitter": undefined,
}
`;

exports[`getPlayer 3`] = `
Object {
  "achievements": Array [],
  "age": 22,
  "country": Object {
    "code": "GE",
    "name": "Georgia",
  },
  "facebook": undefined,
  "id": 3000,
  "ign": "d10",
  "image": "https://static.hltv.org/images/playerprofile/blankplayer.svg",
  "name": "Dati Onezashvili",
  "statistics": Object {
    "deathsPerRound": 0,
    "headshots": 0,
    "killsPerRound": 0,
    "mapsPlayed": 0,
    "rating": 0,
    "roundsContributed": 0,
  },
  "team": undefined,
  "twitch": undefined,
  "twitter": undefined,
}
`;

exports[`getPlayer 4`] = `
Object {
  "achievements": Array [],
  "age": 27,
  "country": Object {
    "code": "US",
    "name": "United States",
  },
  "facebook": undefined,
  "id": 9123,
  "ign": "MAKKA",
  "image": "https://static.hltv.org/images/playerprofile/thumb/9123/400.jpeg?v=2",
  "name": "Bryan Drouillard",
  "statistics": Object {
    "deathsPerRound": 0,
    "headshots": 0,
    "killsPerRound": 0,
    "mapsPlayed": 0,
    "rating": 0,
    "roundsContributed": 0,
  },
  "team": undefined,
  "twitch": undefined,
  "twitter": undefined,
}
`;
